### PR TITLE
Task/ctaa 1435 whats new

### PR DESCRIPTION
**Pull request by nilsgrabenhorst**
Ticket: https://tasks.pxp-x.com/browse/CTAA-1424

Interactive dismiss gesture is still missing. Will have to look into this later.

### CODE REVIEW

File: CoronaContact/Scenes/What's New/AppVersion.swift [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+import UIKit
+
+protocol AppInfo {
+    var appVersion: AppVersion { get }
+}
+
+extension Bundle {
+    var appVersion: String? {
+        return infoDictionary?["CFBundleShortVersionString"] as? String
```
Comment (finnsch) [inline CoronaContact/Scenes/What's New/AppVersion.swift]:
```suggestion
        infoDictionary?["CFBundleShortVersionString"] as? String
```

File: CoronaContact/Scenes/What's New/AppVersion.swift [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    }
+    var appBuild: String? {
+        return infoDictionary?["CFBundleVersion"] as? String
```
Comment (finnsch) [inline CoronaContact/Scenes/What's New/AppVersion.swift]:
```suggestion
        infoDictionary?["CFBundleVersion"] as? String
```

File: CoronaContact/Scenes/What's New/WhatsNewCoordinator.swift [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+import UIKit
+
+final class WhatsNewCoordinator: Coordinator {
+
+    // MARK: - Properties
+
+    private lazy var rootViewController: WhatsNewViewController = {
+        let controller = WhatsNewViewController.instantiate()
+//        controller.viewModel = SavedIDsViewModel(self)
```
Comment (finnsch) [inline CoronaContact/Scenes/What's New/WhatsNewCoordinator.swift]: Do you intend to uncomment this code line later on?

File: CoronaContact/Scenes/What's New/WhatsNewRepository.swift [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+//  WhatsNewRepository.swift
+//  CoronaContact
+//
+
+import UIKit
+
+typealias AppHistoryItem = String
+typealias AppVersionHistory = [AppVersion: AppHistoryItem]
+
+var appVersionHistory: AppVersionHistory = [
```
Comment (finnsch) [inline CoronaContact/Scenes/What's New/WhatsNewRepository.swift]: I would recommend to scope these two properties.

File: CoronaContact/Scenes/What's New/WhatsNewRepository.swift [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+            .map { $0.value }
+    }
+    
+    var isWhatsNewAvailable: Bool {
+        
+        // for the first upgrade to version 2.0 we cannot detect if it is a new
+        // install or an upgrade. We have to show the history item:
+        #warning("Remove this check for the first update after 2.0")
+        if lastWhatsNewShown == .notPreviouslyInstalled && currentAppVersion == firstHistoryItemVersion {
+            // using a very old versuon number in order to show what's new for 2.0:
```
Comment (finnsch) [inline CoronaContact/Scenes/What's New/WhatsNewRepository.swift]:
```suggestion
            // using a very old version number in order to show what's new for 2.0:
```

File: CoronaContact/Scenes/What's New/WhatsNewViewController.swift [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+//  CoronaContact
+//
+
+import UIKit
+import Reusable
+
+class WhatsNewViewController: UIViewController, StoryboardBased {
+    
+    let viewModel = WhatsNewViewModel()
+    weak var coordinator: WhatsNewCoordinator?
```
Comment (finnsch) [inline CoronaContact/Scenes/What's New/WhatsNewViewController.swift]: View controllers should not talk directly to coordinators. This has been handled by the view model so far.

**Review by finnsch — CHANGES_REQUESTED** [review]

File: CoronaContact/Scenes/What's New/AppVersion.swift (hunk 1 above)
Comment (nilsgrabenhorst) [reply CoronaContact/Scenes/What's New/AppVersion.swift]: Fixed

File: CoronaContact/Scenes/What's New/AppVersion.swift (hunk 2 above)
Comment (nilsgrabenhorst) [reply CoronaContact/Scenes/What's New/AppVersion.swift]: Fixed

File: CoronaContact/Scenes/What's New/WhatsNewCoordinator.swift (hunk 3 above)
Comment (nilsgrabenhorst) [reply CoronaContact/Scenes/What's New/WhatsNewCoordinator.swift]: No, sorry, that was a leftover that I forgot to remove. Fixed.

File: CoronaContact/Scenes/What's New/WhatsNewRepository.swift (hunk 5 above)
Comment (nilsgrabenhorst) [reply CoronaContact/Scenes/What's New/WhatsNewRepository.swift]: Fixed

File: CoronaContact/Scenes/What's New/WhatsNewViewController.swift (hunk 6 above)
Comment (nilsgrabenhorst) [reply CoronaContact/Scenes/What's New/WhatsNewViewController.swift]: Sorry about that. I had read the Coordinator Chapter in the objc.io Architecture Book the night before, which sent me on a different path.

Comment (nilsgrabenhorst) [reply CoronaContact/Scenes/What's New/WhatsNewViewController.swift]: Should be fixed now.

File: CoronaContact/Scenes/What's New/WhatsNewRepository.swift (hunk 4 above)
Comment (nilsgrabenhorst) [reply CoronaContact/Scenes/What's New/WhatsNewRepository.swift]: Done

**Review by finnsch — APPROVED** [review]

**Pull request merged**